core/src/main/scala/com/databricks/labs/remorph/parsers/tsql/TSqlExpressionBuilder.scala

```diff
@@ -66,25 +78,83 @@ class TSqlExpressionBuilder
+      // x.y
+      case (c1: ir.Column, c2: ir.Column) =>
+        ir.Column(c1.name + "." + c2.name)
+      // Other cases
+      case _ => ir.Dot(left, right)
+    }
+  }
+
+  override def visitPrimitiveConstant(ctx: TSqlParser.PrimitiveConstantContext): ir.Expression = {
+    if (ctx.DOLLAR != null) {
+      return ir.Literal(string = Some(ctx.getText))
+    }
+    buildConstant(ctx.con)
+  }
+
+  override def visitScNot(ctx: TSqlParser.ScNotContext): ir.Expression =
+    ir.Not(ctx.searchCondition().accept(this))
+
+  override def visitScAnd(ctx: TSqlParser.ScAndContext): ir.Expression =
+    ir.And(ctx.searchCondition(0).accept(this), ctx.searchCondition(1).accept(this))
+
+  override def visitScOr(ctx: TSqlParser.ScOrContext): ir.Expression =
+    ir.Or(ctx.searchCondition(0).accept(this), ctx.searchCondition(1).accept(this))
+
+  override def visitScPred(ctx: TSqlParser.ScPredContext): ir.Expression = ctx.predicate().accept(this)
+
+  override def visitScPrec(ctx: TSqlParser.ScPrecContext): ir.Expression = ctx.searchCondition.accept(this)
+
+  override def visitPredicate(ctx: TSqlParser.PredicateContext): ir.Expression = {
+    val left = ctx.expression(0).accept(this)
+    val right = ctx.expression(1).accept(this)
+
+    ctx.comparisonOperator match {
+      case op if op.LT != null && op.EQ != null => ir.LesserThanOrEqual(left, right)
+      case op if op.GT != null && op.EQ != null => ir.GreaterThanOrEqual(left, right)
+      case op if op.LT != null && op.GT != null => ir.NotEquals(left, right)
+      case op if op.EQ != null => ir.Equals(left, right)
+      case op if op.GT != null => ir.GreaterThan(left, right)
+      case op if op.LT != null => ir.LesserThan(left, right)
+    }
   }
 
-  override def visitTerminal(node: TerminalNode): ir.Expression = node.getSymbol.getType match {
-    case c if c == STRING => Literal(string = Some(removeQuotes(node.getText)))
-    case c if c == INT => Literal(integer = Some(node.getText.toInt))
-    case c if c == FLOAT => Literal(float = Some(node.getText.toFloat))
-    case c if c == HEX => Literal(string = Some(node.getText)) // Preserve format for now
-    case c if c == REAL => Literal(double = Some(node.getText.toDouble))
-    case c if c == NULL_ => Literal(nullType = Some(ir.NullType()))
-    case _ => wrapUnresolvedInput(node.getText)
+  /**
+   * For now, we assume that we are dealing with Column names. Later we can add some context by keeping a symbol table
+   * for DECLARE. LOCAL_ID is not catered for as part of an expression in the current grammar, but even that can be an
+   * alias for a column name.
+   *
+   * For now then, they are all seen as columns.
+   *
+   * @param ctx
+   *   the parse tree
+   */
+  override def visitId(ctx: IdContext): ir.Expression = ctx match {
+    case c if c.ID() != null => ir.Column(ctx.getText)
+    case c if c.TEMP_ID() != null => ir.Column(ctx.getText)
+    case c if c.DOUBLE_QUOTE_ID() != null => ir.Column(ctx.getText)
```

Comment:
we should probably remove the enclosing `"` here (as well as the `[` and `]` bellow)

Reply:
Thought about that, but there is only one isQuoted flag, so I left it for now. Just like the parens, is it important to remember which type of quote was used? I suspect not given that we are producing a different SQL dialect. I will do that in a differnt PR in the interests of getting this one over the line.